### DISCUSSION TITLE
How do we set memory and/or cpu limits when setting up a node?

### DISCUSSION THREAD

**Discussion by migs35323** [k3s-io/k3s#9588] [Q&A] [answered]
what is the "correct" way to setup memory limits in k3s?
in the service files there is the following message:
i am using ubuntu 22 based machines which i believe they come with cgroups enabled
```
# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
```
so, does k3s provide an easy way to set memory\cpu limits node wide?

**Comment by brandond** [comment, marked as answer]
That comment specifically is about the `LimitNOFILE=1048576` entry that follows it, not about the memory or CPU limits.

You can reference the PR that added that comment: https://github.com/k3s-io/k3s/pull/1735

If you want to add CPU or memory limits for the K3s process, you can set them on the lines that follow.
